src/main/java/fact/TriggerEmulation/EmulateLogic.java

```
@@ -0,0 +1,159 @@
+        for (int patch = first_patch_of_ftu; patch < first_patch_of_ftu+Constants.N_PATCHES_FTU; patch++) {
+            ftus.add(patch);
+        }
+        return ftus;
+    }
+
+    /**
+     * Iterate over patches and decide weather the event has triggered for the given trigger properties
+     * @param triggerPrimitives Array with discriminator trigger decisions
+     * @param patchTriggerSlice Array with discriminator trigger times
+     * @param nOutOf4 numper of necessary trigger patches to have a signal
+     * @param nOutOf40 numper of necessary trigger units to have a signal
+     * @param timeWindowSize duration of coincidence of the triggered patches
+     * @return
+     */
+    public static boolean hasTriggered(
+            boolean[] triggerPrimitives,
+            int[] patchTriggerSlice,
+            int nOutOf4,
+            int nOutOf40,
+            int timeWindowSize
+    ) {
+        ArrayList<Integer> triggerTimes = new ArrayList<>();
+
+        int n_units = 0;
+        for (int ftu = 0; ftu < Constants.N_FTUS; ftu++) {
+            int n_trigger_patches = 0;
+
+            for (Integer patch :
+                    getPatchesOfFTU(ftu)) {
+
+                if (triggerPrimitives[patch] && patchTriggerSlice[patch] > 0){
+                    insertToArrayListSorted(patchTriggerSlice[patch], triggerTimes);
+                    n_trigger_patches++;
+                }
+                if (n_trigger_patches >= nOutOf4){
+                    n_units++;
+                    break;
+                }
+            }
+            if (n_units >= nOutOf40){
+                if (areEnoughTriggersInOneTimeWindow(triggerTimes, nOutOf40, timeWindowSize)){
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    /**
+     * sorted insert of int value into an int array
+     * @param x
+     * @param list
+     */
+    public static void insertToArrayListSorted(int x, ArrayList<Integer> list) {
```

This silently ignores the case `>0`, maybe it should throw an error?